Jellyfin.Api/Helpers/AudioHelper.cs

```diff
@@ -154,4 +159,31 @@ public async Task<ActionResult> GetAudioStream(
             transcodingJobType,
             cancellationTokenSource).ConfigureAwait(false);
     }
+
+    /// <summary>
+    /// Gets audio data to draw a waveform bar.
+    /// </summary>
+    /// <param name="itemId">The item id.</param>
+    /// <returns>A <see cref="FileStream"/> containing the resulting .dat file.</returns>
+    public async Task<FileStream> GetAudioWaveForm(Guid itemId)
+    {
+        var item = _libraryManager.GetItemById<BaseItem>(itemId)
+            ?? throw new ResourceNotFoundException();
+
+        var libraryOptions = BaseItem.LibraryManager.GetLibraryOptions(item);
+        var saveInMediaFolder = libraryOptions.SaveLocalMetadata;
+        var saveFileName = Path.GetFileNameWithoutExtension(item.Path) + ".dat";
+        string outputPath = saveInMediaFolder
+            ? Path.Combine(item.ContainingFolderPath, saveFileName)
+            : Path.Combine(item.GetInternalMetadataPath(), saveFileName);
+
+        if (!File.Exists(outputPath))
+        {
+            await Task.Run(() => Process.Start("ffprobe", $"-v error -f lavfi -i \"amovie={item.Path},asetnsamples=44100,astats=metadata=1:reset=1\" -show_entries frame_tags=lavfi.astats.Overall.RMS_level -of csv=p=0 -o \"{outputPath}\"").WaitForExit()).ConfigureAwait(false);
```

Comment:
- 44100 sample batch is only about 1 sample per second and that is too low to generate a good waveform. This value should be dynamic and being adjusted accordingly from the input source audio's sample rate to make each sample has a constant time.
- RMS_level is a bad choice for waveforms. The effective energy level from root mean squares are vary "flat" when plotted, and the squared value is always positive which means this will look differently from the waveforms you see from most audio software, where the wave is from the raw signal which has both positive and negative amplitudes.

Reply:
This is how your waveform is going to look like, as you can see it has a quite boring shape:

![image](https://github.com/user-attachments/assets/5169312f-7868-4310-a27e-ef58006599eb)